src/stores/useUsdcStore.ts

```diff
@@ -0,0 +1,51 @@
+import { Web3Provider } from '@ethersproject/providers';
+import { Web3ReactContextInterface } from '@web3-react/core/dist/types';
+import BigNumber from 'bignumber.js';
+import ERC20Abi from 'config/abi/IERC20.json';
+import { banks } from 'config/constants/banks';
+import { Address } from 'config/constants/types';
+import { getContract } from 'utils/contractHelper';
+import { isLocalhost } from 'utils/misc';
+import createStore from 'zustand';
+import { combine } from 'zustand/middleware';
+
+const initialState = {
+  usdcBalance: 0,
+  isLoading: true,
+};
+
+export const useUsdcStore = createStore(
+  combine(initialState, (set, _get) => ({
+    initialize: async ({
+      chainId,
+      library,
+      account,
+    }: Required<
+      Pick<Web3ReactContextInterface<Web3Provider>, 'chainId' | 'library' | 'account'>
```

Comment:
The use of `Required` and `Pick` types here is noteworthy. If you haven't dug into TypeScript's utility types, I highly recommend you do so: https://www.typescriptlang.org/docs/handbook/utility-types.html